app/services/malware_scanner.rb

```diff
@@ -36,12 +35,16 @@ def virus_found?
     @virus_found ||= !scan_result.status.success?
   end
 
+  def scanner_down?
+    scan_result.status.exitstatus != 2
+  end
+
```

Comment:
I think there's a logic issue here too. My understanding (which could be wrong!) is that an `exitstatus` of 2 indicates an error, so this method should return true if `exitstatus == 2`, not `!=2`.

Alternatively if you rename the method to `scanner_working?` (or `scanner_up?` maybe?) and make the same change on line 21 then there's no need to change the logic.